### PR TITLE
Fix internal http request

### PR DESCRIPTION
**Pull request by chaspy**
ref: #79 
Can't load favorites
screenshot 2019-02-17 17 47 04

### CODE REVIEW

**Comment by chaspy** [comment]
```
prod:
`var url = 'http://favsearch.chaspy.me:80/api/v1.0/favorite';`

stg:
`var url = 'https://favsearch-pr-81.herokuapp.com:443/api/v1.0/favorite';`

local:
`var url = 'http://localhost:4567/api/v1.0/favorite';`
```

File: app.rb [hunk 1]
```diff
@@ -67,9 +67,9 @@
 end
 
 get '/top' do
-  @DomainName = if request.port.nil? && ((request.port == '443') || (request.port == '80'))
```
Comment (chaspy) [inline app.rb]: This condition was not working.

**Comment by chaspy** [comment]
It is now accessible via https://favsearch.chaspy.me by #79. 
However, inside heroku seems to be accessing sinatra (dyno) with `http` .
It should be redirected from http to https on cloudfront, but request seems not to work.

@kamontia 
Please review and merge

File: app.rb [hunk 2]
```diff
@@ -67,9 +67,9 @@
...
-                  request.scheme + '://' + request.host
-                else
+  @DomainName = if request.scheme == 'http' # for production
```
Comment (kamontia) [inline app.rb]: Only this precondition fetches the tweets via `https` protocol in the local environment.
In the local environment,  not `https` but `http` is used, isn't it?

Comment (chaspy) [reply app.rb]: Thanks! addressed c77c182

**Comment by kamontia** [comment]
@chaspy Thanks for your work. It seems that it works fine.

**Pull request merged**